### PR TITLE
[OPENSTACK-2847] Retry when icontrol return 401

### PR DESCRIPTION
**Pull request by zhaoqin-github**
@<reviewer_id>
#### What issues does this address?
Fixes #<issueid>
WIP #<issueid>
...

### CODE REVIEW

File: f5_openstack_agent/lbaasv2/drivers/bigip/system_helper.py [hunk 1]
```diff
@@ -15,40 +15,38 @@
...
 
-        return bigip.tm.sys.folders.folder.exists(name=folder)
+        f = Folder()
+        return f.exists(bigip, name=folder)
 
     def get_folders(self, bigip):
         f_collection = []
-        folders = bigip.tm.sys.folders.get_collection()
+        f = Folder()
+        folders = f.get_collection(bigip)
```
Comment (Niklaus-xie) [inline f5_openstack_agent/lbaasv2/drivers/bigip/system_helper.py]: does not have get_collection() method at all.    maybe you are trying to call get_resources ?   also several other similar calls in this file. e.g. d.get_collection(bigip) in #71, #80, etc.   should modify these.

File: f5_openstack_agent/lbaasv2/drivers/bigip/system_helper.py [hunk 2]
```diff
@@ -69,138 +67,37 @@ def get_minor_version(self, bigip):
         return version
 
     def get_version(self, bigip):
-        devices = bigip.tm.cm.devices.get_collection()
+        d = Device()
+        devices = d.get_collection(bigip)
```
Comment (Niklaus-xie) [inline f5_openstack_agent/lbaasv2/drivers/bigip/system_helper.py]: seems similar here and other places as well . pls check .

File: f5_openstack_agent/lbaasv2/drivers/bigip/system_helper.py (hunk 1 above)
Comment (zhaoqin-github) [reply f5_openstack_agent/lbaasv2/drivers/bigip/system_helper.py]: Done.

File: f5_openstack_agent/lbaasv2/drivers/bigip/system_helper.py (hunk 2 above)
Comment (zhaoqin-github) [reply f5_openstack_agent/lbaasv2/drivers/bigip/system_helper.py]: Done.

**Review by zh805 — APPROVED** [review]

**Pull request merged**